gemd/entity/base_entity.py

```diff
@@ -71,9 +74,36 @@ def add_uid(self, scope, uid):
         """
         self.uids[scope] = uid
 
+    def to_link(self, scope: Optional[str] = None, *, allow_fallback: bool = False) -> LinkByUID:
+        """
+        Generate a LinkByUID for this object.
+
+        Parameters
+        ----------
+        scope: str, optional
+            scope of the uid to get
+        allow_fallback: bool
+            whether to grab another scope/id if chosen scope is missing (Default: False).
+
+        Returns
+        -------
+        LinkByUID
+
+        """
+        if len(self.uids) == 0:
+            raise ValueError(f"{type(self)} {self.name} does not have any uids.")
+
+        if scope is None or allow_fallback and scope not in self.uids:
```

Comment:
I suggest using parentheses to indicate how these logical units are grouped, since (A | B) & C is not the same as A | (B & C).